espflash/src/cli/mod.rs

```diff
@@ -136,15 +136,12 @@ pub struct FlashArgs {
     /// Erase specified data partitions
     #[arg(long, value_name = "PARTS", value_enum, value_delimiter = ',')]
     pub erase_data_parts: Option<Vec<DataType>>,
-    /// Logging format.
-    #[arg(long, short = 'L', default_value = "serial", requires = "monitor")]
-    pub log_format: LogFormat,
     /// Open a serial monitor after flashing
     #[arg(short = 'M', long)]
     pub monitor: bool,
-    /// Baud rate at which to read console output
-    #[arg(long, requires = "monitor", value_name = "BAUD")]
-    pub monitor_baud: Option<u32>,
+    /// Monitor configuration
+    #[clap(flatten)]
+    pub monitor_args: MonitorConfigArgs,
```

Comment:
Not sure how to make this require `monitor` (I coulndt get `requires` to work as it `flatten`), but it doesnt make sense that an user could set the `monitor_args` and not set `monitor`. I can only think of a check in the `flash` method that checks this and throws an error.

Reply:
Hmm yeah not sure how much we can really do about this. There are unfortunately some issues with using `flatten`, but it's a bit of a necessity with how things are currently structured. For now it's probably fine, we can merge as-is and open an issue to investigate this further.

Reply:
Issue created: https://github.com/esp-rs/espflash/issues/764